### PR TITLE
[Merged by Bors] - feat: port Analysis.Calculus.Deriv.Inverse

### PR DESCRIPTION
**Pull request by urkud**
---

- [x] depends on: #4434
- [x] depends on: #4436

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#4434
* leanprover-community/mathlib4#4436
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Analysis/Calculus/Deriv/Inverse.lean [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+/-!
+# Inverse function theorem - the easy half
+
+In this file we prove that `g' (f x) = (f' x)⁻¹` provided that `f` is strictly differentiable at
+`x`, `f' x ≠ 0`, and `g` is a local left inverse of `f` that is continuous at `f x`. This is the
+easy half of the inverse function theorem: the harder half states that `g` exists.
+
+For a more detailed overview of one-dimensional derivatives in mathlib, see the module docstring of
+`analysis/calculus/deriv/basic`.
```
Comment (fpvandoorn) [inline Mathlib/Analysis/Calculus/Deriv/Inverse.lean:22]:
```suggestion
`Analysis/Calculus/Deriv/Basic`.
```

**Review by fpvandoorn — APPROVED** [review]
bors d+

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#4434~~
* ~~leanprover-community/mathlib4#4436~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**